onnx/defs/shape_inference.h

```diff
@@ -44,8 +44,22 @@ inline bool hasExactlyNInputTypes(InferenceContext& ctx, int n, const std::strin
   return true;
 }
 
+inline bool hasNInputShapes(InferenceContext& ctx, int n, const std::string& opname) {
```

Comment:
probably not worth adding a parameter "opname" just for error message (if you really want to show the opname, you can store the opname in the InferenceContext itself).